### PR TITLE
Tbb unload issue

### PR DESCRIPTION
**Pull request by riverlijunjie**
### Details:
Fixing for tbb unload issue
Windows run inference on CPU, tbb library cannot be unloaded, which cause memory leak and unload/lingering threads.

This PR contains:
1. Add a new property to control whether force to terminate tbb threads.
2. Property key is "FORCE_TBB_TERMINATE", default value is false.
3. Explicitly to terminate tbb task scheduler during unload openvino dll if this property is set true.
    e.g: core.set_property(device, ov::force_tbb_terminate(true));
4. If not set FORCE_TBB_TERMINATE, there will be no any additional tbb operations.
5. Remove static FrondEndManager, use a global shared FrondEndManager
6. Replace static ExecutorManager with a global shared ExecutorManager.

### Tickets:

- 82487

### CODE REVIEW

File: src/inference/src/threading/ie_executor_manager.cpp [hunk 1]
```diff
@@ -90,9 +131,55 @@ void ExecutorManagerImpl::clear(const std::string& id) {
...
+    ExecutorManagerHolder() = default;
+    ~ExecutorManagerHolder() = default;
+
+    ExecutorManager::Ptr get(bool addRef) {
+        std::lock_guard<std::mutex> lock(_mutex);
+        if (!_manager) {
+            _manager = std::make_shared<ExecutorManagerImpl>();
+            _refCount = 0;
+        }
+        if (addRef) {
```
Comment (tomdol) [inline src/inference/src/threading/ie_executor_manager.cpp:154]: I don't see any other call of this method with `addRef == false`. Shouldn't it always increase the ref count?

File: src/inference/src/ie_network_reader.hpp [hunk 2]
```diff
@@ -10,6 +10,7 @@
 #include "ie_blob.h"
 #include "ie_iextension.h"
 #include "openvino/core/extension.hpp"
+#include "openvino/frontend/manager.hpp"
```
Comment (tomdol) [inline src/inference/src/ie_network_reader.hpp:13]: This include file is not used in the header, can you remove it or move it to the cpp file if it's required there?

File: src/core/include/openvino/op/constant.hpp [hunk 3]
```diff
@@ -713,6 +713,8 @@ class OPENVINO_API Constant : public Op {
         return shape_size(m_shape) * m_element_type.size();
     }
 
+    friend class frontend::FrontEnd;
+    std::shared_ptr<void> m_shared_object{};
```
[the remark this answers is not in the document]
Comment (riverlijunjie) [reply src/core/include/openvino/op/constant.hpp:717]: PR11236 is a generic, but I found there was segment fault if run this test - CachingTest/CachingTest.LoadHetero_OneDevice/ByModelName_testCache'

File: src/frontends/common/src/frontend.cpp [hunk 4]
```diff
@@ -26,6 +27,11 @@ std::shared_ptr<ov::Model> FrontEnd::create_copy(const std::shared_ptr<ov::Model
                                         ov_model->get_friendly_name());
     copy->m_shared_object = shared_object;
     copy->get_rt_info() = ov_model->get_rt_info();
+    for (auto& op : copy->get_ordered_ops()) {
+        if (auto c = std::dynamic_pointer_cast<ov::op::v0::Constant>(op)) {
+            c->m_shared_object = shared_object;
```
[the remark this answers is not in the document]
Comment (riverlijunjie) [reply src/frontends/common/src/frontend.cpp:32]: yes, I agree with you, maybe we need PR11236 first merged. BTW, this PR is based on PR11315, maybe we can remove it from this PR.

File: src/inference/src/threading/ie_executor_manager.cpp [hunk 5]
```diff
@@ -2,33 +2,74 @@
...
+    clear();
+    std::lock_guard<std::mutex> guard(tbbMutex);
+    if (tbbTerminateFlag) {
+#if IE_THREAD == IE_THREAD_TBB || IE_THREAD == IE_THREAD_TBB_AUTO
+        try {
+            if (_tbb) {
+                _tbb->blocking_terminate();
+            }
+            _tbb = nullptr;
+        } catch (std::exception& e) {
```
[the remark this answers is not in the document]
Comment (riverlijunjie) [reply src/inference/src/threading/ie_executor_manager.cpp:67]: ok, will do it.

File: src/inference/src/threading/ie_executor_manager.cpp (hunk 1 above)
Comment (riverlijunjie) [reply src/inference/src/threading/ie_executor_manager.cpp:154]: There are many places to call executorManager(), which is global, so we don't hope to add refcount at each calling point, only for ov::Core construction we add refCount and decreass refCount in ov::Core destruction.

File: src/inference/src/ie_network_reader.hpp (hunk 2 above)
Comment (riverlijunjie) [reply src/inference/src/ie_network_reader.hpp:13]: sure

Comment (riverlijunjie) [reply src/inference/src/ie_network_reader.hpp:13]: ok

File: src/frontends/common/src/manager.cpp [hunk 6]
```diff
@@ -242,3 +242,51 @@ template <>
 FrontEnd::Ptr FrontEndManager::load_by_model(const std::vector<ov::Any>& variants) {
     return load_by_model_impl(variants);
 }
+
+namespace ov {
+namespace frontend {
+
+namespace {
+
+class FrontEndManagerHolder {
```
[the remark this answers is not in the document]
Comment (riverlijunjie) [reply src/frontends/common/src/manager.cpp:251]: ok

**Comment by ilya-lavrenov** [comment]
@riverlijunjie is https://github.com/openvinotoolkit/openvino/pull/11650 replacement for this PR? can this one be closed then?

**Comment by riverlijunjie** [comment]
> @riverlijunjie is #11650 replacement for this PR? can this one be closed then?

yes, it can be closed.

**Pull request closed without merging**